pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -1380,4 +1365,52 @@ bool ConsumerImpl::isConnected() const { return !getCnx().expired() && state_ ==
 
 uint64_t ConsumerImpl::getNumberOfConnectedConsumer() { return isConnected() ? 1 : 0; }
 
+void ConsumerImpl::seekAsyncInternal(long requestId, SharedBuffer seek, const MessageId& seekId,
+                                     long timestamp, ResultCallback callback) {
+    ClientConnectionPtr cnx = getCnx().lock();
+    if (!cnx) {
+        LOG_ERROR(getName() << " Client Connection not ready for Consumer");
+        callback(ResultNotConnected);
+        return;
+    }
+
+    const auto originalSeekMessageId = seekMessageId_.get();
+    seekMessageId_ = seekId;
+    duringSeek_ = true;
+    if (timestamp > 0) {
+        LOG_INFO(getName() << " Seeking subscription to " << timestamp);
+    } else {
+        LOG_INFO(getName() << " Seeking subscription to " << seekId);
+    }
+
+    std::weak_ptr<ConsumerImpl> weakSelf{shared_from_this()};
+
+    cnx->sendRequestWithId(seek, requestId)
+        .addListener([this, weakSelf, callback, originalSeekMessageId](Result result,
+                                                                       const ResponseData& responseData) {
```

Comment:
We must check if `this` is still valid by validating the reference count of `weakSelf`.

```c++
auto self = weakSelf.lock();
if (!self) {  // this is invalid now
```

Reply:
A weak pointer is captured just to ensure `this` is valid by validating the referenced object is still alive. For example, as you can see

```c++
                ackGroupingTrackerPtr_->flushAndClean();
```

If what `this` points to an invalid object, the access to `ackGroupingTrackerPtr_` could cause segmentation fault.

Reply:
Oh, I see. Thanks for explaining!